shorebird_code_push/lib/src/updater.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:ffi' as ffi;
+
+import 'package:ffi/ffi.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+import 'package:platform/platform.dart';
+import 'package:shorebird_code_push/src/generated/updater_bindings.g.dart';
+
+/// {@template updater}
+/// A wrapper around the generated [UpdaterBindings] that translates ffi types
+/// into easier to use Dart types.
+/// {@endtemplate}
+class Updater {
+  /// Creates an [Updater] instance using the currently loaded dynamic library.
+  Updater.init() {
+    bindings = UpdaterBindings(ffi.DynamicLibrary.process());
+  }
+
+  /// Creates an [Updater] instance using the dynamic library at the given
+  /// [directory].
+  ///
+  /// The name of the library is determined by the current OS:
+  ///   macOS:   lib[name].dylib
+  ///   Windows: [name].dll
+  ///   Linux:   lib[name].so
+  Updater.initWithLibrary({
+    required String directory,
+    required String name,
+    ffi.DynamicLibrary Function(String) ffiOpen =
+        ffi.DynamicLibrary.open, // for testing
+  }) {
+    final dylib = () {
+      if (platform.isMacOS) {
+        return ffiOpen(p.join(directory, 'lib$name.dylib'));
+      }
+      if (platform.isWindows) {
+        return ffiOpen(p.join(directory, '$name.dll'));
+      }
+      // Assume everything else follows the Linux pattern.
+      return ffiOpen(p.join(directory, 'lib$name.so'));
+    }();
+    bindings = UpdaterBindings(dylib);
+  }
+
+  /// The ffi bindings to the Updater library.
+  @visibleForTesting
+  static late UpdaterBindings bindings;
+
+  /// An override for dart:io's [Platform] class for testing.
+  @visibleForTesting
+  static Platform platform = const LocalPlatform();
```

Comment:
Isn't this package included in customer code?  I suspect we want to minimize dependencies as much as possible. 🤷

Reply:
Removed, along with other vestigial Dart CLI code